presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxSplit.java

```diff
@@ -53,7 +53,7 @@ public NodeSelectionStrategy getNodeSelectionStrategy()
         return HARD_AFFINITY;
     }
 
-    @JsonProperty
+    @JsonProperty(value = "addresses")
```

Comment:
Why do we now need to explicitly add the `values`?  Shouldn't this be inferred from the fields themselves?

Reply:
When using @JsonProperty, it's important to explicitly specify the value to ensure Jackson serializes the field with the exact name you want in the JSON. Jackson automatically changes field names, especially for boolean fields. When a method starts with is (like isDirectMapped()), Jackson assumes the field name is directMapped, not isDirectMapped.

For example, in the case with the isDirectMapped() method:

>  @JsonProperty(value = "isDirectMapped")
>         public Optional<Boolean> isDirectMapped()
>         {
>             return isDirectMapped;
>         }

The @JsonProperty("isDirectMapped") annotation ensures Jackson uses isDirectMapped as the field name in the JSON, rather than inferring it and removing the "is" prefix. This guarantees the correct field name is serialized in the JSON.